src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.h

```diff
@@ -61,6 +61,14 @@ class SmokeCoAlarmServer
      */
     bool SetExpressedState(chip::EndpointId endpointId, ExpressedStateEnum newExpressedState);
 
+    /**
+     * @brief Set the highest level of Expressed State according to priorityOrder
+     * @param endpointId ID of the endpoint
+     * @param priorityOrder Priority order of expressed state from highest to lowest
+     * @return true on success, false on failure
+     */
+    bool AutoSetExpressedState(chip::EndpointId endpointId, std::array<ExpressedStateEnum, kPriorityOrderLength> priorityOrder);
```

Comment:
should array be passed in as a `const &` ?

Reply:
It's changed.